Services/Calendar/classes/class.ilCalendarSchedule.php

```diff
@@ -106,26 +106,29 @@ public function __construct(ilDate $seed,$a_type,$a_user_id = 0, $a_strict_perio
 	 					
 		
 		// category / event filters
-		
-		include_once('./Services/Calendar/classes/class.ilCalendarCategories.php');
-		
+
 		// portfolio does custom filter handling (booking group ids)		
 		if(ilCalendarCategories::_getInstance()->getMode() != ilCalendarCategories::MODE_PORTFOLIO_CONSULTATION)
 		{
 			// consultation hour calendar views do not mind calendar category visibility
 			if(ilCalendarCategories::_getInstance()->getMode() != ilCalendarCategories::MODE_CONSULTATION)
 			{
 				// this is the "default" filter which handles currently hidden categories for the user
-				include_once('./Services/Calendar/classes/class.ilCalendarScheduleFilterHidden.php');
-				$this->addFilter(new ilCalendarScheduleFilterHidden($this->user->getId()));		
+				$this->addFilter(new ilCalendarScheduleFilterHidden($this->user->getId()));
 			}
 			else
 			{
 				// handle booking visibility (target object, booked out)
-				include_once('./Services/Calendar/classes/class.ilCalendarScheduleFilterBookings.php');
-				$this->addFilter(new ilCalendarScheduleFilterBookings($this->user->getId()));		
+				//this filter deals with consultation hours
+				$this->addFilter(new ilCalendarScheduleFilterBookings($this->user->getId()));
 			}
-			
+
+			if(ilCalendarCategories::_getInstance()->getMode() === ilCalendarCategories::MODE_PERSONAL_DESKTOP_MEMBERSHIP)
```

Comment:
This is very restrictive and modifies the booking pool events only for one personal desktop type.
But I guess the events should be manipulated also for the other category types. Please try to add this filter directly in the "else" branch.

Reply:
Yes, I see it, such ilCalendarCategories modes are a bit confusing for me. 
In this case, the booking pool events are always != ilCalendarCategories::MODE_CONSULTATION
I think the filter should be inside de "if" instead of the else isn't it?